### PR TITLE
chore (litmus-portal): Refactor Settings section to use InputField

### PR DESCRIPTION
**Pull request by saswatamcode**
## Proposed changes

Refactored all the components inside of the `Settings` section to use the unified and customisable `InputField` component. This will ensures consistency in styling as well as input handling and validation methods. 

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Special notes for your reviewer:
The `UserManagement` component still uses material-ui `TextField` for Search input due to unique requirements.

### CODE REVIEW

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 1]
```diff
@@ -55,9 +45,20 @@ const AccountSettings: React.FC = () => {
     err: false,
   });
 
+  // handleCurrPassword handles password validation for first password field
+  const handleCurrPassword = (prop: keyof Password) => (
+    event: React.ChangeEvent<{ value: string }>
+  ) => {
+    setCurrPassword({
+      ...currPassword,
+      err: false,
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:54]: You don't need `err` the validation can handle it. You can remove `err` from all the password local states

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 2]
```diff
@@ -139,99 +117,34 @@ const AccountSettings: React.FC = () => {
...
-                <InputLabel htmlFor="outlined-adornment-password">
-                  New Password
-                </InputLabel>
-              </FormControl>
+              <InputField
+                required
+                type="password"
+                handleChange={handleNewPassword('password')}
+                label="New Password"
+                validationError={newPassword.err}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:135]: Make `validationError={validateLength(newPassword.password)}` here

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 3]
```diff
@@ -139,99 +117,34 @@ const AccountSettings: React.FC = () => {
...
-                <InputLabel htmlFor="outlined-adornment-password">
-                  Confirm new password
-                </InputLabel>
-              </FormControl>
+              <InputField
+                required
+                type="password"
+                handleChange={handleConfPassword('password')}
+                label="Confirm Password"
+                validationError={confNewPassword.err}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:145]: Make `validationError={validateConfirmPassword(
                        newPassword.password,
                        confNewPassword.password
           )}`

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 4]
```diff
@@ -139,99 +117,34 @@ const AccountSettings: React.FC = () => {
...
-                        )}
-                      </IconButton>
-                    </InputAdornment>
-                  }
-                />
-                <InputLabel htmlFor="outlined-adornment-password">
-                  Confirm new password
-                </InputLabel>
-              </FormControl>
+              <InputField
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:140]: Add a `helperText` here and make it
```
helperText={
                      validateConfirmPassword(
                        newPassword.password,
                        confNewPassword.password
                      )
                        ? 'Password is not same'
                        : ''
                    }
```

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 5]
```diff
@@ -55,9 +45,20 @@ const AccountSettings: React.FC = () => {
...
+      ...currPassword,
+      err: false,
+      [prop]: event.target.value,
+    });
+  };
+
   // handleNewPassword handles password validation for second password field
   const handleNewPassword = (prop: keyof Password) => (
-    event: React.ChangeEvent<HTMLInputElement>
+    event: React.ChangeEvent<{ value: string }>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:61]: Change all the {value: string} to HTMLInputElement

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
 import userAvatar from '../../../../../../utils/user';
 import useStyles from './styles';
+import {
+  validateOnlyAlphabet,
+  validateEmail,
+} from '../../../../../../utils/validate';
 
 interface PersonalDetailsProps {
-  handleNameChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleNameChange?: (event: React.ChangeEvent<{ value: string }>) => void;
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx:12]: Change them to `HTMLInputElement`

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx [hunk 7]
```diff
@@ -53,45 +58,31 @@ const UserDetails: React.FC<PersonalDetailsProps> = ({
...
-              id="filled-user-input"
+              handleChange={handleNameChange}
+              validationError={validateOnlyAlphabet(nameValue)}
               label="Full Name"
-              InputProps={{ disableUnderline: true }}
-              data-cy="fullName"
             />
 
-            <TextField
+            <InputField
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx:70]: Add a helper text here too based on the validation, take reference from `Stepper.tsx`

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx [hunk 8]
```diff
@@ -53,45 +58,31 @@ const UserDetails: React.FC<PersonalDetailsProps> = ({
           </div>
           {/* Fields for details including Full name, email, username */}
           <div className={classes.details1}>
-            <TextField
+            <InputField
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/UserDetails/index.tsx:61]: Add a helper text here too based on the validation, take reference from `Stepper.tsx`

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx [hunk 9]
```diff
@@ -51,7 +41,7 @@ const CreateUser: React.FC<CreateUserProps> = ({ handleDiv }) => {
 
   // handles password field
   const handleCreatePassword = (prop: keyof Password) => (
-    event: React.ChangeEvent<HTMLInputElement>
+    event: React.ChangeEvent<{ value: string }>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx:44]: Value to HTMLElement

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx [hunk 10]
```diff
@@ -149,53 +133,21 @@ const CreateUser: React.FC<CreateUserProps> = ({ handleDiv }) => {
...
                     <div className={classes.details1}>
-                      <TextField
-                        className={classes.userDetail}
-                        id="filled-username-input"
+                      <InputField
                         label="Username"
-                        defaultValue="RichardHill"
                         value={personalData.userName}
+                        handleChange={handleUsername}
+                        validationError={false}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx:140]: Validation here can be `validateStartEmptySpacing`

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx [hunk 11]
```diff
@@ -149,53 +133,21 @@ const CreateUser: React.FC<CreateUserProps> = ({ handleDiv }) => {
...
-                        <InputLabel htmlFor="outlined-adornment-password">
-                          New Password
-                        </InputLabel>
-                      </FormControl>
+                      <InputField
+                        required
+                        type="password"
+                        handleChange={handleCreatePassword('password')}
+                        value={createPAssword.password}
+                        validationError={createPAssword.err}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/CreateUser/index.tsx:148]: The password validation can be made here like suggested above

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/EditUser/index.tsx [hunk 12]
```diff
@@ -122,41 +100,14 @@ const EditUser: React.FC<EditUserProps> = ({
...
-                        />
-                        <InputLabel htmlFor="outlined-adornment-password">
-                          New Password
-                        </InputLabel>
-                      </FormControl>
+                      <InputField
+                        required
+                        handleChange={handleCreatePassword('password')}
+                        type="password"
+                        validationError={createPAssword.err}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/EditUser/index.tsx:107]: The password validation can be made here like suggested above with validateLength

File: litmus-portal/frontend/src/components/WelcomeModal/Stepper.tsx [hunk 13]
```diff
@@ -312,7 +312,7 @@ const CStepper: React.FC<CStepperProps> = ({ handleModal }) => {
                     validationError={false}
                     value={values.password}
                     success={isSuccess.current}
-                    handleChange={(event) =>
+                    handleChange={(event: { target: { value: any } }) =>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/WelcomeModal/Stepper.tsx:315]: Change all these to HTMLInputElement

File: litmus-portal/frontend/src/containers/layouts/InputField/index.tsx [hunk 14]
```diff
@@ -13,16 +13,18 @@ interface InputFieldProps {
   helperText?: string;
   validationError: boolean;
   success?: boolean;
+  disabled?: boolean;
   value: string;
-  required: boolean;
+  required?: boolean;
   iconType?: string | undefined;
-  handleChange: (event: React.ChangeEvent<{ value: string }>) => void;
+  handleChange?: (event: React.ChangeEvent<{ value: string }>) => void;
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/containers/layouts/InputField/index.tsx:20]: handleChange cannot be an optional parameter, input fields always change and receive an event and change it to HTMLInputElement

File: litmus-portal/frontend/src/pages/LoginPage/index.tsx [hunk 15]
```diff
@@ -79,7 +79,7 @@ const LoginPage = () => {
                 validationError={validateStartEmptySpacing(authData.username)}
                 data-cy="inputEmail"
                 required
-                handleChange={(e) =>
+                handleChange={(e: { target: { value: any } }) =>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/pages/LoginPage/index.tsx:82]: Remove this, you are type casting the event to be of type any whereas it is automatically infering the type to `React.ChangeEvent<HTMLInputElement>`

File: litmus-portal/frontend/src/pages/LoginPage/index.tsx [hunk 16]
```diff
@@ -99,7 +99,7 @@ const LoginPage = () => {
                 // validationError={validatePassword(authData.password)}
                 validationError={false}
                 data-cy="inputPassword"
-                handleChange={(e) =>
+                handleChange={(e: { target: { value: any } }) =>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/pages/LoginPage/index.tsx:102]: Change this too and keep it `e` only

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 17]
```diff
@@ -139,99 +117,34 @@ const AccountSettings: React.FC = () => {
...
-                  }
-                />
-                <InputLabel htmlFor="outlined-adornment-password">
-                  Confirm new password
-                </InputLabel>
-              </FormControl>
+              <InputField
+                required
+                type="password"
+                handleChange={handleConfPassword('password')}
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:143]: Add a `success` field here, you can use validateConfirmPassword and negate it or use useRef. Take a look at `Stepper` password validation for reference.

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 18]
```diff
@@ -139,99 +117,34 @@ const AccountSettings: React.FC = () => {
...
-                        )}
-                      </IconButton>
-                    </InputAdornment>
-                  }
-                />
-                <InputLabel htmlFor="outlined-adornment-password">
-                  New Password
-                </InputLabel>
-              </FormControl>
+              <InputField
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:130]: Add a `success` field here, you can use validateConfirmPassword and negate it or use useRef. Take a look at `Stepper` password validation for reference.

**Review by S-ayanide — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/src/containers/layouts/InputField/index.tsx [hunk 19]
```diff
@@ -13,16 +13,18 @@ interface InputFieldProps {
   helperText?: string;
   validationError: boolean;
   success?: boolean;
+  disabled?: boolean;
   value: string;
-  required: boolean;
+  required?: boolean;
   iconType?: string | undefined;
-  handleChange: (event: React.ChangeEvent<{ value: string }>) => void;
+  handleChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/containers/layouts/InputField/index.tsx:20]: Make handleChange a required parameter. For Search on Settings use InputBase like its in `BrowseWorkflow/HeaderSection.tsx`. Search does not need validation

**Review by S-ayanide — CHANGES_REQUESTED** [review]

**Comment by S-ayanide** [comment]
# Few problems -
1. The new password field is green by default because it is based on validation check which is true initially as well when both are blank, since blank === blank is true. Changing the confirm new password would toggle the success check (green border).
2. Password is too short helper text is visible by default
3. Previous Password is in error state by default

**Review by S-ayanide — APPROVED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by S-ayanide — APPROVED** [review]

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx [hunk 20]
```diff
@@ -139,106 +93,60 @@ const AccountSettings: React.FC = () => {
...
+                handleChange={handleConfPassword('password')}
+                success={isSuccess.current}
+                label="Confirm Password"
+                validationError={validateConfirmPassword(
+                  newPassword.password,
+                  confNewPassword.password
+                )}
+                value={confNewPassword.password}
+              />
               <Button
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/AccountSettings/index.tsx:144]: Custom Button can be used?

**Pull request merged**